pkg/scheduler/scheduler.go

```diff
@@ -155,28 +159,28 @@ func (pc *Scheduler) loadSchedulerConf() {
 }
 
 func (pc *Scheduler) watchSchedulerConf(stopCh <-chan struct{}) {
-	if pc.fileWatcher == nil {
+	informerFactory := informers.NewSharedInformerFactory(pc.cache.Client(), 0)
+	cmInformer := informerFactory.Core().V1().ConfigMaps()
+	cmInformer.Informer().AddEventHandler(
+		cache.ResourceEventHandlerFuncs{
+			UpdateFunc: pc.updateConfigMap,
+		})
+	informerFactory.Start(stopCh)
+}
+
+// updateConfigMap update scheduler config
+func (pc *Scheduler) updateConfigMap(_, newObj interface{}) {
+	cm, ok := newObj.(*v1.ConfigMap)
+	if !ok {
+		klog.Errorf("Cannot convert to *v1.ConfigMap: %v", newObj)
 		return
 	}
-	eventCh := pc.fileWatcher.Events()
-	errCh := pc.fileWatcher.Errors()
-	for {
-		select {
-		case event, ok := <-eventCh:
-			if !ok {
-				return
-			}
-			klog.V(4).Infof("watch %s event: %v", pc.schedulerConf, event)
-			if event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Create == fsnotify.Create {
-				pc.loadSchedulerConf()
-			}
-		case err, ok := <-errCh:
-			if !ok {
-				return
-			}
-			klog.Infof("watch %s error: %v", pc.schedulerConf, err)
-		case <-stopCh:
-			return
-		}
+	if cm.GetNamespace() != "volcano-system" && cm.GetName() != pc.schedulerConfigmap {
```

Comment:
hard coded "volcano-system" may not a good idea, users may use other ns like `kube-system`